server/src/main/java/com/gitittogether/skillForge/server/controller/user/UserController.java

```
@@ -0,0 +1,97 @@
+        return ResponseEntity.ok(response);
+    }
+
+    /**
+     * Fetches the profile of a user by their ID.
+     *
+     * @param userId The ID of the user whose profile is to be fetched.
+     * @return ResponseEntity with the user's profile information.
+     */
+    @GetMapping("/{userId}/profile")
+    public ResponseEntity<?> getUserProfile(@PathVariable String userId) {
+        log.info("🔍 Fetching profile for user ID: {}", userId);
+        return ResponseEntity.ok(userService.getUserProfile(userId));
+    }
+
+    /**
+     * Updates the profile of a user.
+     *
+     * @param userId  The ID of the user whose profile is to be updated.
+     * @param request The update request containing new profile details.
+     * @return ResponseEntity with the updated user's profile information.
+     */
+    @PutMapping("/{userId}/profile")
+    public ResponseEntity<?> updateUserProfile(@PathVariable String userId, @Valid @RequestBody UserProfileUpdateRequest request) {
+        log.info("🔄 Updating profile for user ID: {}", userId);
+        return ResponseEntity.ok(userService.updateUserProfile(userId, request));
+    }
+
+    /**
+     * Deletes a user profile by their ID.
+     *
+     * @param userId The ID of the user to be deleted.
+     * @return ResponseEntity indicating the result of the deletion operation.
+     */
+    @DeleteMapping("/{userId}/profile")
+    public ResponseEntity<?> deleteUserProfile(@PathVariable String userId) {
+        log.info("🗑️ Deleting profile for user ID: {}", userId);
+        boolean deleted = userService.deleteUser(userId);
+        if (!deleted) {
+            log.warn("❌ Failed to delete user profile for ID: {}", userId);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("User not found");
+        } else {
+            log.info("✅ Successfully deleted user profile for ID: {}", userId);
+            return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
```

The deleteUserProfile method has conditional response status codes, but the service method only returns boolean. Consider enhancing the service method to throw exceptions instead of returning boolean for more consistent error handling across the application.